src/EnableByMatchingSegment.php

```diff
@@ -58,7 +63,12 @@ public function toArray(): array
     }
 
     /**
-     * @return array<string, string|array>
+     * @return array<string, string|array<string, string|array<string, mixed>>>
+     * @psalm-return array{
+     *   id: string,
+     *   segments: array<array-key, array<string, array<array-key, mixed>|string>>,
+     *   type: "enable_by_matching_segment"
```

Comment:
Maybe we could set the type string instead of the value. What do you think?

Reply:
Is the type given by psalm, it adds value because is a constant ;-D

Reply:
We can do better by defining a custom type for Psalm/PHPStan ;-D

Reply:
@xserrat I updated the PR to use Psalm types. I think it looks much better now. It will require the next min0r release the v0.2.5 of toggle-model package before merge ;-D

Reply:
Perfect @kpicaza!! It looks simple and consistent!👏🏽👏🏽